packages/view-types/spatial/src/Spatial.js

```diff
@@ -708,6 +715,17 @@ class Spatial extends AbstractSpatialOrScatterplot {
     }
   }
 
+  resizeScaleBar() {
+    if (this.state.scaleBarSizeFactor > 1.5) {
+      return;
+    }
+    this.setState({ scaleBarSizeFactor: this.state.scaleBarSizeFactor + 0.1 }, () => {
+      setTimeout(() => {
+        this.setState({ scaleBarSizeFactor: this.state.scaleBarSizeFactor - 0.1 });
+      }, 40);
```

Comment:
Is this setTimeout needed? DeckGL is typically performant enough to do computations on every render

Reply:
I added it, because otherwise the difference in bar length would not be noticeable at all. From what I understand, this is not because of deck.gl, but because of React's handling of state changes.